library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2990,9 +2977,9 @@ impl<T, A: Allocator> Extend<T> for VecDeque<T, A> {
 }
 
 #[stable(feature = "extend_ref", since = "1.2.0")]
-impl<'a, T: 'a + Copy, A: Allocator> Extend<&'a T> for VecDeque<T, A> {
+impl<'a, T: 'a + Copy, A: Allocator + 'a> Extend<&'a T> for VecDeque<T, A> {
```

Comment:
I may have blindly copied it from the `Vec` implementation :sweat_smile:

Reply:
Ok, can you remove it? It might be a correct addition but such things should be fixed in a separate PR and reviewed by someone who knows more about allocators.

Reply:
Actually I can't. I just tried and it says

```
error[E0309]: the parameter type `A` may not live long enough
    --> library/alloc/src/collections/vec_deque/mod.rs:2994:9
     |
2994 |         self.spec_extend(iter.into_iter());
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = help: consider adding an explicit lifetime bound `A: 'a`...

For more information about this error, try `rustc --explain E0309`.
error: could not compile `alloc` due to previous error
```

If I then also remove the lifetime from `A` in `SpecExtend` it says

```
error: cannot specialize on `Binder(OutlivesPredicate(A, ReEarlyBound(0, 'a)), [])`
  --> library/alloc/src/collections/vec_deque/spec_extend.rs:60:1
   |
60 | / impl<'a, T: 'a, A: Allocator + 'a> SpecExtend<&'a T, slice::Iter<'a, T>> for VecDeque<T, A>
61 | | where
62 | |     T: Copy,
63 | | {
...  |
72 | |     }
73 | | }
   | |_^

error: could not compile `alloc` due to previous error
```

Reply:
This shouldn't be an issue though, right? Allocator is an unstable API, so it shouldn't break anything

Reply:
You have to remove the `+ 'a` on both `SpecExtend<&'a T, ...>` impls